gwpy/tests/test_timeseries.py

```diff
@@ -1154,6 +1154,18 @@ def test_rms(self, losc):
         rms = losc.rms(1.)
         assert rms.sample_rate == 1 * units.Hz
 
+    def test_demod(self):
+        # create a timeseries with two loud lines with different phases
+        # at different frequencies
+        amp, phase, f = 1., 0., 30
+        t = numpy.linspace(0, 10, 10*16384)
+        data = amp * numpy.cos(2*numpy.pi*f*t + phase)
+        data = TimeSeries(data, unit='strain', times=t)
+        assert data.demod(f).unit == data.unit
+        assert len(data.demod(f)) == 10
+        assert all(x <= 1e-4 for x in numpy.abs(data.demod(f).value - amp))
+        assert all(x <= 1e-4 for x in (numpy.angle(data.demod(f)) - phase))
```

Comment:
@alurban, is the purpose of these `all` calls to assert that the result is close to zero? If so it makes more sense to me to use something like

```python
        data = TimeSeries(amp * numpy.cos(2*numpy.pi*f*t + phase),
                                        unit='', times=t)
        demod = data.demod(f)
        assert demod.size == 10
        assert demod.unit == data.unit
        utils.assert_allclose(numpy.abs(demod.value), amp)
        utils.assert_allclose(numpy.angle(demod.value), phase)
```

Reply:
@duncanmmacleod you're right that this is exactly what I was going for. I was unaware of `assert_allclose` before, it seems perfect for this test. I coded that up in this commit: https://github.com/gwpy/gwpy/pull/705/commits/1977ee56063d3b2fc1c4955ce253738353745736